### PR TITLE
[ISSUE #3495] Optimize the eventmesh-admin of EventMesh

### LINKED ISSUE #3495: [Enhancement] Optimize the eventmesh-admin of EventMesh
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

At present, eventmesh-admin provides a management interface for eventmesh storage, but it only implements the management function of rocketmq, which needs to be further expanded. At the same time, it can provide CLI for users to quickly start and experience. 

### Describe the solution you'd like

At present, eventmesh-admin provides a management interface for eventmesh storage, but it only implements the management function of rocketmq, which needs to be further expanded. At the same time, it can provide CLI for users to quickly start and experience. 

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by fabian4**
Fixes #3495.

### Modifications

- refactor with new module `eventmesh-admin-api`.
- add `eventmesh-admin-redis`.
- fit `eventmesh-admin-rocketmq`.

### CODE REVIEW

File: eventmesh-admin/eventmesh-admin-rocketmq/src/main/java/org/apache/eventmesh/admin/rocketmq/service/RocketMQService.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public RocketMQService() {
+        this.redisAdmin = StoragePluginFactory.getMeshMQAdmin(PLUGIN_NAME);
+    }
+
+    @Override
+    public TopicResponse createTopic(String topic) {
+        TopicResponse topicResponse = null;
+        try {
+            redisAdmin.createTopic(topic);
+            topicResponse = new TopicResponse(topic,
```
Comment (Pil0tXia) [inline eventmesh-admin/eventmesh-admin-rocketmq/src/main/java/org/apache/eventmesh/admin/rocketmq/service/RocketMQService.java:46]: Please check after copying the code.

**Review by Pil0tXia — COMMENTED** [review]
The admin module is designed to be deployed independently, and it's better not to use SPI. It's also recommended to avoid heavy reliance on storage.api in admin.api. The existing interface path matching mechanism is not as good as the `@EventHttpHandler` annotation in `runtime.admin`. Therefore, the current design will pose significant difficulty for subsequent work.

It will be a good idea to add new features after the refactoring is complete. You may take a look at other modules first, like httpRetry. Currently, the retry logic for each protocol is separate, and we can create an API there to reduce coupling.

**Comment by fabian4** [comment]
> The admin module is designed to be deployed independently, and it's better not to use SPI. It's also recommended to avoid heavy reliance on storage.api in admin.api. The existing interface path matching mechanism is not as good as the `@EventHttpHandler` annotation in `runtime.admin`. Therefore, the current design will pose significant difficulty for subsequent work.
> 
> It will be a good idea to add new features after the refactoring is complete. You may take a look at other modules first, like httpRetry. Currently, the retry logic for each protocol is separate, and we can create an API there to reduce coupling.

Should I keep this pr until the refactoring is done ? If so let me know when it is ready.

**Comment by Pil0tXia** [comment]
@fabian4 I'll get back to you when the refactor is complete, and please feel free to do anything.

**Comment by fabian4** [comment]
> @fabian4 please fix the conflicts.

@qqeasonchen https://github.com/apache/eventmesh/pull/4405 is ready to merge.

This one may still be hold when @Pil0tXia is ready.

**Review by weihubeats — COMMENTED** [review]
please fix the conflicts.

**Comment by Pil0tXia** [comment]
@qqeasonchen @weihubeats The dependencies introduced by this PR do not align with the design of the admin module, so it cannot be merged at this time.

**Comment by Pil0tXia** [comment]
@fabian4 The EventMesh Dashboard is under devlopment in https://github.com/apache/eventmesh-dashboard Repo, and you're welcome to come and join us~

**Pull request closed without merging**